### PR TITLE
Add API allowing to check if captured event is ANR error 

### LINKED ISSUE #574: Expose more sentry types to Unreal or expose an delegate for listening ANR
What problem could Sentry solve that it doesn't?

Currently, if I want to filter ANR in beforesend handler, I have to use multiple if-condition together like `!Event->IsCrash()` and `Event->GetLevel() == ESentryLevel::Error` and possibly `Event->GetMessage().StartWith(TEXT("App Hanging"))` to mimic the same conditions set in https://github.com/getsentry/sentry-cocoa/blob/05896990562960d04052bf5a1548e290d89228bf/Sources/Sentry/SentryANRTrackingIntegration.m#L87-L103 

And it's not accurate due to I don't have SentryException and SentryMechanism exposed to Unreal, nothing returns me a SentryANRExceptionType with a USentryEvent pointer.

Maybe you could expose those above,  OR give another handler specifically tailored with ANR event?

### PR DESCRIPTION
**Pull request by tustanivsky**
This PR adds a new method to `USentryEvent` which allows checking if the captured event is an ANR error. By default, ANR error tracking is turned off in plugin settings.

Note, that ANR error tracking works only on Android/Apple since `sentry-native` doesn't support it at the moment.

Closes #574

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 ## Unreleased
 
+### Features
+
+- Add API allowing to check if captured event is ANR error ([#581](https://github.com/getsentry/sentry-unreal/pull/581))
```
Comment (bitsandfoxes) [inline CHANGELOG.md:7]:
```suggestion
- Added `IsAnr()` method to mobile `SentryEvents` to check if the event is caused by the Application Not Responding ([#581](https://github.com/getsentry/sentry-unreal/pull/581))
```

File: plugin-dev/Source/Sentry/Public/SentryEvent.h [hunk 2]
```diff
@@ -48,6 +48,10 @@ class SENTRY_API USentryEvent : public UObject
 	UFUNCTION(BlueprintPure, Category = "Sentry")
 	bool IsCrash() const;
 
+	/** Gets flag indicating whether the event is an Application Not Responding (ANR) error. */
+	UFUNCTION(BlueprintPure, Category = "Sentry")
+	bool IsAppNotResponding() const;
```
Comment (bitsandfoxes) [inline plugin-dev/Source/Sentry/Public/SentryEvent.h:53]: It looks like it's nice having it written out in the blueprints?

**Review by bitsandfoxes — APPROVED** [review]

File: plugin-dev/Source/Sentry/Public/SentryEvent.h (hunk 2 above)
Comment (tustanivsky) [reply plugin-dev/Source/Sentry/Public/SentryEvent.h:53]: We can add a `meta=(DisplayName="Is App Not Responding")` attribute for UFUNCTION here in order to display the full name in blueprints while still having `IsAnr` in C++ as everywhere else. Does that sound good?

Comment (bitsandfoxes) [reply plugin-dev/Source/Sentry/Public/SentryEvent.h:53]: That'd be great!

**Pull request merged**